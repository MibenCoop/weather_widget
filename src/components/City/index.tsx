import React from "react";

export type Props = {
  name: string;
  temperature: number;
};

const City = ({ name, temperature }: Props) => (
  <p>
    City name: {name}, Temperature: {temperature + 1}
  </p>
);
export default City;
