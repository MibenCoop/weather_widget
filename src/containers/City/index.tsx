import { connect } from 'react-redux';
import React, { Component } from 'react';

import CityComponent from '../../components/City';

import { CityWeather } from '../../interfaces';
import Spinner from '../../components/Spinner';
import { fetchWeather } from '../../actions/cityWeather';

type cityProps = {
    name: string;
};

export class CityWeatherList extends Component<cityProps> {
    componentDidMount() {
        const { name, fetchWeather }: any = this.props;
        fetchWeather(name);
    }
    componentDidUpdate(prevProps: any) {
        const { name, fetchWeather }: any = this.props;
        if (prevProps.name !== name) fetchWeather(name);
    }
    render() {
        const { name, isFetching, temperature }: any = this.props;
        if (isFetching) return <Spinner />;
        return <CityComponent name={name} temperature={temperature} />;
    }
}

type CityWeatherState = {
    citiesWeather: CityWeather;
};

const mapStateToProps = ({ citiesWeather }: CityWeatherState) => ({
    temperature: citiesWeather.temperature,
    isFetching: citiesWeather.isFetching,
});

export default connect(mapStateToProps, {
    fetchWeather: (name: string) => fetchWeather(name),
})(CityWeatherList);
