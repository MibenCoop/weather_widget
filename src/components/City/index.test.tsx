import React from "react";
import { shallow, mount } from "enzyme";

import City from "./index";

describe("City weather component", () => {
  const cityWeather = {
    temperature: 10,
    name: "Moscow"
  };
  it("should match snapshot", () => {
    const city = shallow(
      <City name={cityWeather.name} temperature={cityWeather.temperature} />
    );
    expect(city).toMatchSnapshot();
  });
  it("should be called with props name: city and value: Moscow", () => {
    const city = mount(
      <City name={cityWeather.name} temperature={cityWeather.temperature} />
    );
    expect(city.prop("name")).toEqual(cityWeather.name);
  });
  it("should be called with props name: temperature and value: 10", () => {
    const city = mount(
      <City name={cityWeather.name} temperature={cityWeather.temperature} />
    );
    console.log("(city.length", city.length);
    expect(city.prop("temperature")).toEqual(cityWeather.temperature);
  });
});
