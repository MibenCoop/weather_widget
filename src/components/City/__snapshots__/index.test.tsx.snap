// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City weather component should match snapshot 1`] = `
<p>
  City name: 
  Moscow
  , Temperature: 
  11
</p>
`;
